src/textual/_ansi_sequences.py

```diff
@@ -367,7 +295,109 @@
+            # Arrows + Shift
+            "\x1b[a": (Keys.ShiftUp,),
+            "\x1b[b": (Keys.ShiftDown,),
+            "\x1b[c": (Keys.ShiftRight,),
+            "\x1b[d": (Keys.ShiftLeft,),
+            # Arrows + Meta
+            "\x1b\x1b[A": (Keys.Escape, Keys.Up),
+            "\x1b\x1b[B": (Keys.Escape, Keys.Down),
+            "\x1b\x1b[C": (Keys.Escape, Keys.Right),
+            "\x1b\x1b[D": (Keys.Escape, Keys.Left),
+            "\x1b\x1b[8~": (Keys.Escape, Keys.End),
+            "\x1b\x1b[7~": (Keys.Escape, Keys.Home),
+            # Arrows + Meta + Shift
+            "\x1b\x1b[a": (Keys.Escape, Keys.ShiftUp),
+            "\x1b\x1b[b": (Keys.Escape, Keys.ShiftDown),
+            "\x1b\x1b[c": (Keys.Escape, Keys.ShiftRight),
+            "\x1b\x1b[d": (Keys.Escape, Keys.ShiftLeft),
+            "\x1b\x1b[8$": (Keys.Escape, Keys.ShiftEnd),
+            "\x1b\x1b[7$": (Keys.Escape, Keys.ShiftHome),
+            # Arrows + Control
+            "\x1bOa": (Keys.ControlUp,),
+            "\x1bOb": (Keys.ControlDown,),
+            "\x1bOc": (Keys.ControlRight,),
+            "\x1bOd": (Keys.ControlLeft,),
+            # Arrows + Control + Meta
+            "\x1b\x1bOa": (Keys.Escape, Keys.ControlUp),
+            "\x1b\x1bOb": (Keys.Escape, Keys.ControlDown),
+            "\x1b\x1bOc": (Keys.Escape, Keys.ControlRight),
+            "\x1b\x1bOd": (Keys.Escape, Keys.ControlLeft),
+            "\x1b\x1b[8^": (Keys.Escape, Keys.ControlEnd),
+            "\x1b\x1b[7^": (Keys.Escape, Keys.ControlHome),
+            # Function keys (F5-Fn work normally)
+            "\x1b[11~": (Keys.F1,),
+            "\x1b[12~": (Keys.F2,),
+            "\x1b[13~": (Keys.F3,),
+            "\x1b[14~": (Keys.F4,),
+            # Function keys + Meta
+            "\x1b\x1b[11~": (Keys.Escape, Keys.F1),
+            "\x1b\x1b[12~": (Keys.Escape, Keys.F2),
+            "\x1b\x1b[13~": (Keys.Escape, Keys.F3),
+            "\x1b\x1b[14~": (Keys.Escape, Keys.F4),
+            "\x1b\x1b[15~": (Keys.Escape, Keys.F5),
+            "\x1b\x1b[17~": (Keys.Escape, Keys.F6),
+            "\x1b\x1b[18~": (Keys.Escape, Keys.F7),
+            "\x1b\x1b[19~": (Keys.Escape, Keys.F8),
+            "\x1b\x1b[20~": (Keys.Escape, Keys.F9),
+            "\x1b\x1b[21~": (Keys.Escape, Keys.F10),
+            "\x1b\x1b[23~": (Keys.Escape, Keys.F11),
+            "\x1b\x1b[24~": (Keys.Escape, Keys.F12),
+        }
+    )
+
+# Linux console
+elif _TERM.startswith("linux"):
+    ANSI_SEQUENCES_KEYS.update(
+        {
+            "\x1b[[A": (Keys.F1,),
+            "\x1b[[B": (Keys.F2,),
+            "\x1b[[C": (Keys.F3,),
+            "\x1b[[D": (Keys.F4,),
+            "\x1b[[E": (Keys.F5,),
+        }
+    )
+
+# tmux
+elif _TERM.startswith("tmux"):
+    ANSI_SEQUENCES_KEYS.update(
+        {
+            "\x1b[1~": (Keys.Home,),
+            "\x1b[4~": (Keys.End,),
+        }
+    )
+
+# TODO: Does Windows Terminal set TERM?
```

Comment:
Windows terminal doesn't, but other terminals on Windows might.

![Screenshot 2023-10-30 at 15 30 50](https://github.com/Textualize/textual/assets/28237/b93d6101-ad74-4855-80e5-336a15fcfc1a)

Above is Windows Terminal, Hyper and WezTerm.